### DISCUSSION TITLE
DNS server configuration in WSL2

### DISCUSSION THREAD

**Discussion by wcndave** [ddev/ddev#6830] [Q&A] [answered]
I have installed ddev with drupal, and have come back to it a week later, and I cannot get any network connectivity to work.

apt-get update and ddev composer both cannot resolve domains.

I have tried to edit /run/systemd/resolve/stub-resolv.conf
and change nameserver to 8.8.8.8 as the 127.0.0.53 didn't work - which helped on a different computer, however resolv.conf kept reverting every few seconds, but I was able to see that it was a name server issue

However changing the nameserver in my current project doesn't help at all, so basically I am stuck and can't do anything...

Can anyone suggest what I can do?

**Comment by rfay** [comment, marked as answer]
Hi @wcndave - We really have to have a lot more information. Please take a look at the [troubleshooting](https://ddev.readthedocs.io/en/stable/users/usage/troubleshooting/) docs, then come back with:

* The file output by `ddev debug test`. That will help us understand what your OS and environment are
* The exact symptom you're facing. Please say what you mean by "cannot get any network connectivity to work"

**Comment by wcndave** [comment]
Really sorry, I discovered after continuing to look that when my VPN is on, this happens (names do not resolve to IPs), however with VPN off, everything is fine.  You can just delete this thread, unless it might be useful for someone else!

**Reply by rfay** [reply]
Glad you got it working!

**Reply by wcndave** [reply]
well, I say that... however there's still an issue.
When I start ddev, it always has the nameserver 127.0.0.11
so I cannot do any package updates or installs.
I change the stub file that is the symlink, and then it sort of works, but composer won't work
then I do ddev ssh, and check resolv.conf and that is also set to 127.0.0.11 so I update that
However then the drupal site cannot find the database... it must be using that even though the db is local...
So I have to add the nameserver 8.8.8.8 to that file.
every time I start ddev?

**Reply by rfay** [reply]
Unfortunately, you're apparently dealing with the configuration of your own Windows machine and your own environment. 

But the WSL docs explain how to make your resolv.conf changes permanent, https://learn.microsoft.com/en-us/windows/wsl/wsl-config#network-settings 

However, you probably should be able to make the DNS config for your whole computer more sane. Or if you control the router, fix the router.

**Reply by wcndave** [reply]
Thanks!  Although I am not sure that my config is not sane.... I have no problems accessing anything at all from my computer, never had a problem resolving anything...  However as you say, probably not a ddev issues, apart from having to change the resolv file in both the wsl container and the ddev container... but I will figure it out.  Thanks again!

**Reply by rfay** [reply]
I've set up many, many windows machines and not seen what you're having, and many thousands of users do this without this problem, so it's about your Windows or network configuration, but you can solve it a few ways, and you're on track.
